FYI, I've tried to use the current state of this PR and get the following compilation error:

```
In file included from /home/jacob/ws/ros1_bridge_ws/install/class_loader/include/class_loader/class_loader_core.hpp:56:0,
                 from /home/jacob/ws/ros1_bridge_ws/install/class_loader/include/class_loader/class_loader.hpp:55,
                 from /home/jacob/ws/ros1_bridge_ws/install/class_loader/include/class_loader/multi_library_class_loader.hpp:52,
                 from /home/jacob/ws/ros1_bridge_ws/install/pluginlib/include/pluginlib/class_loader.hpp:58,
                 from /home/jacob/ws/rosbag2_bag_v2_ws/install/include/rosbag/bag.h:67,
                 from /home/jacob/ws/rosbag2_bag_v2_ws/src/rosbag2_bag_v2/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.hpp:23,
                 from /home/jacob/ws/rosbag2_bag_v2_ws/src/rosbag2_bag_v2/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.cpp:15:
/home/jacob/ws/ros1_bridge_ws/install/class_loader/include/class_loader/meta_object.hpp: In instantiation of ‘B* class_loader::impl::MetaObject<C, B>::create() const [with C = rosbag2_bag_v2_plugins::RosbagV2Storage; B = rosbag2_storage::storage_interfaces::ReadOnlyInterface]’:
/home/jacob/ws/rosbag2_bag_v2_ws/src/rosbag2_bag_v2/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.cpp:185:1:   required from here
/home/jacob/ws/ros1_bridge_ws/install/class_loader/include/class_loader/meta_object.hpp:193:12: error: invalid new-expression of abstract class type ‘rosbag2_bag_v2_plugins::RosbagV2Storage’
     return new C;
            ^~~~~
In file included from /home/jacob/ws/rosbag2_bag_v2_ws/src/rosbag2_bag_v2/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.cpp:15:0:
/home/jacob/ws/rosbag2_bag_v2_ws/src/rosbag2_bag_v2/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.hpp:29:7: note:   because the following virtual functions are pure within ‘rosbag2_bag_v2_plugins::RosbagV2Storage’:
 class RosbagV2Storage : public rosbag2_storage::storage_interfaces::ReadOnlyInterface
       ^~~~~~~~~~~~~~~
In file included from /home/jacob/ws/ros1_bridge_ws/install/rosbag2_storage/include/rosbag2_storage/storage_interfaces/read_only_interface.hpp:20:0,
                 from /home/jacob/ws/rosbag2_bag_v2_ws/src/rosbag2_bag_v2/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.hpp:22,
                 from /home/jacob/ws/rosbag2_bag_v2_ws/src/rosbag2_bag_v2/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.cpp:15:
/home/jacob/ws/ros1_bridge_ws/install/rosbag2_storage/include/rosbag2_storage/storage_interfaces/base_info_interface.hpp:41:23: note: 	virtual std::__cxx11::string rosbag2_storage::storage_interfaces::BaseInfoInterface::get_relative_path() const
   virtual std::string get_relative_path() const = 0;
                       ^~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/rosbag2_bag_v2_plugins.dir/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/rosbag2_bag_v2_plugins.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2
```

--- a/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.cpp
+++ b/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.cpp
@@ -119,6 +119,11 @@ uint64_t RosbagV2Storage::get_bagfile_size() const
   return rosbag2_storage::FilesystemHelper::get_file_size(ros_v2_bag_->getFileName());
 }
 
+std::string RosbagV2Storage::get_relative_path() const
+{
+  return rosbag2_storage::FilesystemHelper::get_file_name(ros_v2_bag_->getFileName());
+}
+
 rosbag2_storage::BagMetadata RosbagV2Storage::get_metadata()
 {
   auto bag_view = std::make_unique<rosbag::View>(*ros_v2_bag_);
@@ -126,9 +131,7 @@ rosbag2_storage::BagMetadata RosbagV2Storage::get_metadata()
   rosbag2_storage::BagMetadata metadata;
   metadata.storage_identifier = get_storage_identifier();
   metadata.bag_size = get_bagfile_size();
-  metadata.relative_file_paths = {
-    rosbag2_storage::FilesystemHelper::get_file_name(full_file_path)
-  };
+  metadata.relative_file_paths = {get_relative_path()};
   metadata.duration = std::chrono::nanoseconds(
     bag_view->getEndTime().toNSec() - bag_view->getBeginTime().toNSec());
   metadata.starting_time = std::chrono::time_point<std::chrono::high_resolution_clock>(

--- a/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.hpp
+++ b/rosbag2_bag_v2_plugins/src/rosbag2_bag_v2_plugins/storage/rosbag_v2_storage.hpp
@@ -45,6 +45,8 @@ public:
 
   uint64_t get_bagfile_size() const override;
 
+  std::string get_relative_path() const override;
+
   /**
    * Returns the identifier for the rosbag_v2 plugin.
    * \returns the identifier "rosbag_v2"

